app/ensure_admin_user.go

```diff
@@ -0,0 +1,143 @@
+package app
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/docker/docker/pkg/reexec"
+	"github.com/pkg/errors"
+	"github.com/rancher/rancher/pkg/api/customization/authn"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/urfave/cli"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func RegisterEnsureDefaultAdminCommand() {
+	reexec.Register("/usr/bin/ensure-default-admin", ensureDefaultAdmin)
+	reexec.Register("ensure-default-admin", ensureDefaultAdmin)
+}
+
+func ensureDefaultAdmin() {
+	app := cli.NewApp()
+	app.Description = "Ensure an available default admin user"
+
+	app.Action = func(c *cli.Context) error {
+		kubeConfigPath := os.ExpandEnv("$HOME/.kube/config")
+		if _, err := os.Stat(kubeConfigPath); err != nil {
+			kubeConfigPath = ""
+		}
+
+		conf, err := clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+		if err != nil {
+			return fmt.Errorf("Couldn't get kubeconfig. %v", err)
+		}
+
+		client, err := v3.NewForConfig(*conf)
+		if err != nil {
+			return errors.Errorf("Couldn't get kubernetes client. %v", err)
+		}
+
+		set := labels.Set(map[string]string{"authz.management.cattle.io/bootstrapping": "admin-user"})
+		admins, err := client.Users("").List(v1.ListOptions{LabelSelector: set.String()})
+		if err != nil {
+			return errors.Errorf("Couldn't get default admin user. %v", err)
+		}
+
+		count := len(admins.Items)
+		if count > 1 {
+			var users []string
+			for _, u := range admins.Items {
+				users = append(users, u.Name)
+			}
+			return errors.Errorf("%v users were found with %v label. They are %v. Can only reset the default admin password when there is exactly one user with this label.",
```

Comment:
Bad copy paste of error message from the reset password logic?

Reply:
Hrm sort of yeah. In this case, there are multiple default admins found (who may or may not be enabled or remain actual admins. 

Do we just want to do it for all of them in that case?  

Obviously there _shouldn't_ be more than one default admin, but we're accommodating for that case in reset-password, so I'm guessing it's come up at some point. I guess technically it would just be multiple users who got _labeled_ in a way indicating them as the default admin.